src/include/mpir_datatype.h

```diff
@@ -575,13 +575,9 @@ int MPIR_Get_elements_x_impl(MPI_Count * bytes, MPI_Datatype datatype, MPI_Count
 void MPIR_Type_get_extent_x_impl(MPI_Datatype datatype, MPI_Count * lb, MPI_Count * extent);
 void MPIR_Type_get_true_extent_x_impl(MPI_Datatype datatype, MPI_Count * true_lb,
                                       MPI_Count * true_extent);
-void MPIR_Get_count_impl(const MPI_Status * status, MPI_Datatype datatype, MPI_Aint * count);
 int MPIR_Type_contiguous_x_impl(MPI_Count count, MPI_Datatype old_type, MPI_Datatype * new_type_p);
-void MPIR_Type_get_extent_impl(MPI_Datatype datatype, MPI_Aint * lb, MPI_Aint * extent);
-void MPIR_Type_get_true_extent_impl(MPI_Datatype datatype, MPI_Aint * true_lb,
-                                    MPI_Aint * true_extent);
 void MPIR_Type_free_impl(MPI_Datatype * datatype);
-void MPIR_Pack_size_impl(int incount, MPI_Datatype datatype, MPI_Aint * size);
+void MPIR_Pack_size(MPI_Aint incount, MPI_Datatype datatype, MPI_Aint * size);
```

Comment:
I'm not sure exactly what's happening in this commit. It seems datatype `_impl` functions are becoming wrappers where they weren't before.
For example:
(old) MPI_Pack_size -> MPIR_Pack_size_impl
(new) MPI_Pack_size -> MPIR_Pack_size -> MPIR_Pack_size_impl
Is this so small and large (_c) versions can be treated differently before calling the _impl API?

Reply:
> 
> 
> I'm not sure exactly what's happening in this commit. It seems datatype `_impl` functions are becoming wrappers where they weren't before.
> For example:
> (old) MPI_Pack_size -> MPIR_Pack_size_impl
> (new) MPI_Pack_size -> MPIR_Pack_size -> MPIR_Pack_size_impl
> Is this so small and large (_c) versions can be treated differently before calling the _impl API?

In the large branch we may swap the array or may not, depending on whether `MPI_Aint` is the same size as `MPI_Count`. When we do the swap, the parameter has to be renamed. This becomes awkward for inlined "body of routines" since to have two `large_count` versions with different variable names seem just silly. Thus it seems much easier to just move the inline code block into `_impl` wrapper, which can be handled automatically.

Reply:
I see. Can you add this more detailed description to the commit message? At the moment
```
datatype: use MPI_Aint for internal impl functions
```
lacks information about the addition of the wrappers.

Reply:
Fixed.